app/uk/gov/hmrc/soletraderidentificationfrontend/controllers/DetailsNotFoundController.scala

```diff
@@ -39,6 +42,8 @@ class DetailsNotFoundController @Inject()(mcc: MessagesControllerComponents,
       authorised() {
         journeyService.getJourneyConfig(journeyId).map {
           journeyConfig =>
+            val remoteMessagesApi = messagesHelper.getRemoteMessagesApi(journeyConfig)
+            implicit val messages: Messages = remoteMessagesApi.preferred(request)
             Ok(view(
               pageConfig = journeyConfig.pageConfig,
               redirectLocation = routes.RetryJourneyController.tryAgain(journeyId),
```

Comment:
not related to this page but messages implementation is missed in details-did-not-match page